messaging/src/main/java/org/axonframework/eventhandling/AbstractEventBus.java

```diff
@@ -110,9 +115,14 @@ public Registration registerDispatchInterceptor(
 
     @Override
     public void publish(@Nonnull List<? extends EventMessage<?>> events) {
-        List<MessageMonitor.MonitorCallback> ingested = events.stream()
-                                                              .map(messageMonitor::onMessageIngested)
-                                                              .collect(Collectors.toList());
+        Span span = spanFactory.createInternalSpan("EventBus.publish", events.get(0))
```

Comment:
Would this mean the span for all the events references the name of the first event? 
That would create incorrect naming for the other events within the batch to publish, right?

Differently put, wouldn't we want a span per event, similarly as is done with the monitor callbacks?

Reply:
This has since been changed. We now create a small span for each ingested event to visualize the application of an event. When committing the unit of work storing these events is then in a commit trace.